marimbabot_behavior/src/marimbabot_behavior/behavior_node.py

```diff
@@ -151,69 +225,110 @@ def callback_command(self, command_msg):
+                    if 'play' in command and self.planning_client.simple_state == actionlib.SimpleGoalState.DONE:
+                        self.play()
+
+                    # if the preview is not busy, the note sequence is send to the lilypond_audio action server
+                    if 'preview' in command and self.lilypond_audio_client.simple_state == actionlib.SimpleGoalState.DONE:
+                        self.preview()
             else:
                 rospy.logwarn('No notes to play. Say reed to read notes.')
                 self.response_pub.publish('No notes to play. Say reed to read notes.')
 
         # play faster or slower than the current tempo (by specified bpm value, default = 20)
-        elif re.match(r'marimbabot play (faster|slower)( by [0-9]+ bpm)?', command):
+        elif re.match(r'marimbabot (play|preview) (faster|slower)( by [0-9]+ bpm)?', command):
             # check if a note sequence has been read via the 'read' command
             if self.note_sequence:
                 value = int(command.split(' ')[-2]) if 'by' in command else 20
-                self.change_tempo(faster=True if 'faster' in command else False, value=value)
-                self.play()
+                result = self.change_tempo(faster=True if 'faster' in command else False, value=value)
+                if result == 'success':
+                    # if the motion is not busy, the hit sequence is send to the planning action server 
+                    if 'play' in command and self.planning_client.simple_state == actionlib.SimpleGoalState.DONE:
+                        self.play()
+
+                    # if the preview is not busy, the note sequence is send to the lilypond_audio action server
+                    if 'preview' in command and self.lilypond_audio_client.simple_state == actionlib.SimpleGoalState.DONE:
+                        self.preview()
             else:
                 rospy.logwarn('No notes to play. Say reed to read notes.')
                 self.response_pub.publish('No notes to play. Say reed to read notes.')
 
-        # preview using sound interpreted by computer (MIDI)
-        elif re.match(r'marimbabot preview( in [0-9]+ bpm)?', command):
-            # if a note sequence has been read via the 'read' command
-            if self.note_sequence:
-                if 'bpm' in command:
-                    value = int(command.split(' ')[-2])
-                    self.assign_tempo(value)
-                self.preview()
-            else:
-                rospy.logwarn('No notes to preview. Say reed to read notes.')
-                self.response_pub.publish('No notes to preview. Say reed to read notes.')
-
-        # preview faster or slower than the current tempo (by specified bpm value, default = 20)
-        elif re.match(r'marimbabot preview (faster|slower)( by [0-9]+)?', command):
+        elif re.match(r'marimbabot (play|preview) (louder|softer)( by [0-9]+ steps)?', command):
```

Comment:
For the coming version of the command recognition, the keyword `marimbabot` is not contained. because the STT is triggered after the keyword `marimbabot`.

If you do need it I and add it manually.